src/Components/Join/Join.jsx

```diff
@@ -27,6 +27,12 @@ class Join extends React.Component {
         document.title = 'Deeper Blue: Join Game';
         this.httpGetRequest('/gamelist');
         this.initAuthListener();
+
+        if (typeof this.props.location.state !== 'undefined') {
```

Comment:
I don't know what this is doing

Reply:
https://reacttraining.com/react-router/web/api/Redirect/to-object

Basically if the user redirects from the `Redirect` component on the `Invite` component:

```jsx
<Redirect to={{
    pathname: "/join",
    state: {
        error: `Couldn't join game ${this.state.gameID}.`
    }
}}/>
```

Then `this.props.location.state.error` can be used to access the error message.

However, if the user opens the `/join` page without having gone through the `Redirect` on the `Invite` component, then `this.props.location.state` would obviously be undefined, since it is only defined by the `Invite` component. 

---

This just ensures that the error is only displayed in the event of a redirect, and doesn't error otherwise.

Guess I should add a little comment maybe.